src/components/Swipeable.js

```diff
@@ -0,0 +1,221 @@
+import { setNextSubject } from '../actions/subject'
+
+//needs to be absolutely positioned below the question, otherwise it covers up
+//everything above and prevents it from being touchable
+const toTop = (Platform.OS === 'ios') ? 135 : 125
+const SWIPE_THRESHOLD = 90
+const leftOverlayColor = theme.swipeLeft
+const rightOverlayColor = theme.swipeRight
+
+const mapStateToProps = (state, ownProps) => ({
+  workflow: state.classifier.workflow[ownProps.workflowID] || {},
+  subject: state.classifier.subject[ownProps.workflowID] || {},
+  classification: state.classifier.classification[ownProps.workflowID] || {},
+  annotations: state.classifier.annotations[ownProps.workflowID] || {},
+  subjectSizes: state.classifier.subjectSizes[ownProps.workflowID] || {},
+  seenThisSession: state.classifier.seenThisSession[ownProps.workflowID] || [],
+  questionContainerHeight: state.classifier.questionContainerHeight[ownProps.workflowID] || 0,
+})
+
+const mapDispatchToProps = (dispatch) => ({
+  saveAnnotation(task, value) {
+    dispatch(saveAnnotation(task, value))
+  },
+  saveThenStartNewClassification(answerIndex) {
+    dispatch(saveThenStartNewClassification(answerIndex))
+  },
+  setNextSubject() {
+    dispatch(setNextSubject())
+  },
+})
+
+export class Swipeable extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      pan: new Animated.ValueXY(),
+      enterAnim: new Animated.Value(.9),
+    }
+  }
+
+  isSwipeGesture(gestureState) {
+    return gestureState.dx > 5 || gestureState.dx < -5
+  }
+
+  componentWillMount() {
+    this._panResponder = PanResponder.create({
+    onStartShouldSetResponder: (evt, gestureState) => this.isSwipeGesture(gestureState),
+    onMoveShouldSetPanResponder: (evt, gestureState) => this.isSwipeGesture(gestureState),
+
+    onPanResponderGrant: () => {
+      this.state.pan.setOffset({x: this.state.pan.x._value, y: this.state.pan.y._value})
+      this.state.pan.setValue({x: 0, y: 0})
+    },
+
+    onPanResponderMove: Animated.event([
+      null, {dx: this.state.pan.x, dy: this.state.pan.y + 100},
+    ]),
+
+    onPanResponderRelease: (e, {vx, vy}) => {
+      this.state.pan.flattenOffset()
+      let velocity
+
+      if (vx >= 0) {
+        velocity = clamp(3, 4, vx)
+      } else if (vx < 0) {
+        velocity = clamp(3, 4, vx * -1) * -1
+      }
+
+      if (Math.abs(this.state.pan.x._value) > SWIPE_THRESHOLD) {
+        //negative=left (no = index 1)
+        //positive=right (yes = index 0)
+        const answer = ( this.state.pan.x._value < 0 ? 1 : 0 )
+        Animated.decay(this.state.pan, {
+          velocity: {x: velocity, y: vy},
+          deceleration: 0.98
+        }).start(this.onAnswered(answer))
+      } else {
+        Animated.spring(this.state.pan, {
+          toValue: {x: 0, y: 0},
+          friction: 4
+        }).start()
+      }
+    }
+    })
+  }
+
+  onAnswered = (answer) => {
```

Comment:
Is there a reason this is using the arrow function? It's not really a big deal and I'm probably being too pedantic but was wondering if it could be written like the other functions.